### DISCUSSION TITLE
Support for list of string/int/float/etc

### DISCUSSION THREAD

**Discussion by thijstriemstra** [forntoh/LcdMenu#29] [Q&A] [answered]
PR #22 is a stab at list support for strings and ints (which works quite well on esp32) but that implementation has drawbacks like usage of vector.

Right now input and toggle are only supported menu types and hopefully lists can be added as well.

**Comment by forntoh** [comment]
I haven't visited the project for a while, I was just getting used to my code (asking myself, did I write this? 😂).

Now, I clearly understand what you mean by support for lists 😎, and I have some ideas.

* When the user selects a List Menu Item.
* Calling `enter()` should cycle through the list _(e.g. just like how we set time on a digital watch)_
* The callback linked to the Menu Item is called on every step.

**Reply by thijstriemstra** [reply]
yea sounds what I had in mind as well.

I'm now using a rotary encoder to control the menu. An `menu.enter()` selects an item (sets `isOn` to true) and then the rotary encoder cycles through the list, depending on the direction you turn the encoder. Whenever an item is selected using encoder rotation, I apply the value instantly (e.g. change color of led). Pressing the rotary encoder again, 'closes' the menu item list selection and allows you to scroll through the rest of the menu.

**Reply by thijstriemstra** [reply]
https://user-images.githubusercontent.com/305679/123260532-f5274680-d4f5-11eb-8de4-cc2439120d02.MOV

**Reply by thijstriemstra** [reply]
in the video you also see 'temp 22' with some garbled text. This is `setText` trying to display a custom degrees celsius char but obviously failing. This would be another enhancement request, add custom char support to setText.

**Reply by forntoh** [reply]
Coming soon 😉 #35

**Comment by forntoh** [comment, marked as answer]
https://lcdmenu.forntoh.dev/examples/string-list
https://lcdmenu.forntoh.dev/examples/int-and-float-values

#114
